source/features/show-names.tsx

```diff
@@ -18,56 +18,60 @@ function dropExtraCopy(element: HTMLAnchorElement): void {
 	}
 }
 
-// The selector observer calls this function several times, but we want to batch them into a single GraphQL API call
-async function updateLink(batchedUsernameElements: HTMLAnchorElement[]): Promise<void> {
-	const myUsername = getUsername();
-	batchedUsernameElements = batchedUsernameElements.filter(({textContent: name}) => name !== myUsername && name !== 'ghost');
-	const usernames = new Set(batchedUsernameElements.map(element => element.textContent));
+function appendName(element: HTMLAnchorElement, fullName: string): void {
+	// If it's a regular comment author, add it outside <strong> otherwise it's something like "User added some commits"
+	const {parentElement} = element;
+	const insertionPoint = parentElement!.tagName === 'STRONG' ? parentElement! : element;
+
+	insertionPoint.after(
+		' ',
+		<span className="color-fg-muted css-truncate d-inline-block">
+			(<bdo className="css-truncate-target" style={{maxWidth: '200px'}}>{fullName}</bdo>)
+		</span>,
+		' ',
+	);
+}
 
-	if (usernames.size === 0) {
+async function updateLink(found: HTMLAnchorElement[]): Promise<void> {
+	// eslint-disable-next-line no-use-extend-native/no-use-extend-native -- Sigh
+	const users = Map.groupBy(found, element => element.textContent.trim());
+	users.delete(getUsername()!);
+	users.delete('ghost'); // Consider using `github-reserved-names` if more exclusions are needed
+
+	if (users.size === 0) {
 		return;
 	}
 
 	const names = await api.v4(
-		[...usernames].map(user =>
-			api.escapeKey(user) + `: user(login: "${user}") {name}`,
+		[...users.keys()].map(username =>
+			api.escapeKey(username) + `: user(login: "${username}") {name}`,
 		).join(','),
 	);
 
-	for (const usernameElement of batchedUsernameElements) {
-		const username = usernameElement.textContent;
+	for (const [username, elements] of users) {
 		const userKey = api.escapeKey(username);
-		const {name} = names[userKey];
+		const {name: fullName} = names[userKey];
 
 		// Could be `null` if not set
-		if (!name) {
-			continue;
-		}
-
-		if (isUsernameAlreadyFullName(username, name)) {
-			usernameElement.textContent = name;
+		if (!fullName) {
 			continue;
 		}
 
-		// If it's a regular comment author, add it outside <strong> otherwise it's something like "User added some commits"
-		const {parentElement} = usernameElement;
-		const insertionPoint = parentElement!.tagName === 'STRONG' ? parentElement! : usernameElement;
-
-		// This change is ideal but should not break the feature if it fails
-		// And it should be done before inserting the name
-		try {
-			dropExtraCopy(usernameElement);
```

Comment:
This was not being called for `isUsernameAlreadyFullName` since https://github.com/refined-github/refined-github/pull/7666

In reality we'd want to just call it as soon as possible, regardless of the replacement, so I moved its call much higher